### PR TITLE
chore: fix warning logs

### PR DESCRIPTION
**Pull request by njtran**
Fixes #N/A 

**Description**
Getting warnings like these all throughout the controller logs, since our template resources in the NodePool were removed. 
```
    logger.go:146: 2024-08-01T12:51:20.830-0700	INFO	KubeAPIWarningLogger	log/warning_handler.go:65	unknown field "spec.template.spec.resources"
```
**How was this change tested?**
testing here

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/apis/v1/nodeclaim.go [hunk 1]
```diff
@@ -75,6 +75,78 @@ type NodeClaimSpec struct {
...
+		StartupTaints:          in.StartupTaints,
+		Requirements:           in.Requirements,
+		NodeClassRef:           in.NodeClassRef,
+		TerminationGracePeriod: in.TerminationGracePeriod,
+		ExpireAfter:            in.ExpireAfter,
+	}
+}
+
+// NodeClaimSpec describes the desired state of the NodeClaim
+type NodeClaimSpecWithoutResources struct {
```
Comment (jonathan-innis) [inline pkg/apis/v1/nodeclaim.go:101]: Consider calling this NodeClaimTemplateSpec and storing this in the NodePool.go file

Comment (njtran) [reply pkg/apis/v1/nodeclaim.go]: Good call

File: pkg/controllers/provisioning/scheduling/scheduling_benchmark_test.go [hunk 2]
```diff
@@ -144,7 +144,7 @@ func benchmarkScheduler(b *testing.B, instanceCount, podCount int) {
 	nodePoolWithMinValues := test.NodePool(v1.NodePool{
 		Spec: v1.NodePoolSpec{
 			Template: v1.NodeClaimTemplate{
-				Spec: v1.NodeClaimSpec{
+				Spec: v1.NodeClaimSpecWithoutResources{
```
Comment (jonathan-innis) [inline pkg/controllers/provisioning/scheduling/scheduling_benchmark_test.go:147]: Missed update here

File: pkg/apis/v1/nodepool.go [hunk 3]
```diff
@@ -160,6 +160,77 @@ func (l Limits) ExceededBy(resources v1.ResourceList) error {
...
+	// +kubebuilder:default:="720h"
+	// +kubebuilder:validation:Pattern=`^(([0-9]+(s|m|h))+)|(Never)$`
+	// +kubebuilder:validation:Type="string"
+	// +kubebuilder:validation:Schemaless
+	// +optional
+	ExpireAfter NillableDuration `json:"expireAfter,omitempty"`
+}
+
+// This is used to convert between the NodeClaim's NodeClaimSpec to the Nodepool NodeClaimTemplate's NodeClaimSpec.
+func (in *NodeClaimTemplateSpec) ToNodeClaimSpec() *NodeClaimSpec {
```
Comment (jonathan-innis) [inline pkg/apis/v1/nodepool.go:221]: Do we need these helpers? When do we need to convert between them?

File: pkg/apis/v1/nodeclaim.go [hunk 4]
```diff
@@ -75,6 +75,18 @@ type NodeClaimSpec struct {
 	ExpireAfter NillableDuration `json:"expireAfter,omitempty"`
 }
 
+// This is used to convert between the Nodepool NodeClaimTemplate's NodeClaimSpec to the NodeClaim's NodeClaimSpec.
+func (in *NodeClaimSpec) ToNodeClaimTemplateSpec() *NodeClaimTemplateSpec {
```
Comment (jonathan-innis) [inline pkg/apis/v1/nodeclaim.go:79]: Same question here. When do we need to convert between these?

File: pkg/apis/v1/nodepool.go (hunk 3 above)
Comment (njtran) [reply pkg/apis/v1/nodepool.go:221]: This one is used to convert the NodePool's nodeclaimTemplate spec to the nodeClaimSpec when we're reasoning about theoretical node claims and represent the resource requests in the nodeclaim.

File: pkg/apis/v1/nodeclaim.go (hunk 4 above)
Comment (njtran) [reply pkg/apis/v1/nodeclaim.go:79]: This actually isn't needed, removed.

File: pkg/apis/v1/nodepool.go [hunk 5]
```diff
@@ -160,6 +160,77 @@ func (l Limits) ExceededBy(resources v1.ResourceList) error {
...
+		StartupTaints:          in.StartupTaints,
+		Requirements:           in.Requirements,
+		NodeClassRef:           in.NodeClassRef,
+		TerminationGracePeriod: in.TerminationGracePeriod,
+		ExpireAfter:            in.ExpireAfter,
+	}
+}
+
+// NodeClaimTemplateWithResources is used in scheduling simulations
+type NodeClaimTemplateWithResources struct {
```
Comment (jonathan-innis) [inline pkg/apis/v1/nodepool.go:233]: Do we need this if we are just going to convert to a NodeClaim that can track resources?

Comment (jonathan-innis) [reply pkg/apis/v1/nodepool.go:233]: Why not just store the Resources in the scheduling.go file? Leaking the scheduling implementation into the APIs package feels wrong to me

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**